### PR TITLE
Various cleanups & fixes

### PR DESCRIPTION
**Pull request by vigo2**
[core] remove no longer used ReducedXXXHitTakenChance pseudo stats
[core] make ThunderClapAura "classic"
[core] fix racials, e.g. all reduced hit taken chances are now just 10 resistance or 1% dodge, ranged crit becomes ranged weapon skill, humans gain 5% spirit (up from 3%), and tauren 5% health (up from just base health)

[druid] Hurricane cannot crit anymore

[hunter] Aimed Shot no longer benefits from Barrage [hunter] Carve, Raptor Strike, Serpent Sting, and traps no longer benefit from Mortal Shots

[mage] Combustion no longer increases crit multiplier

[shaman] correct lower Flurry ranks
[shaman] Way of Earth now increases all threat by 50%

[warrior] simplify warrior critical strike damage
[warrior] One-Handed Weapon Specialization now increases damage by 2% per point 
[warrior] Thunder Clap now deals physical damage (reduced by armor), but using defense type "magic" (i.e. OutcomeMagicHitAndCrit and SpellCritMultiplier); ThreatMultiplier increased to 2.5 w/o, and 3.75 w/ Furious Thunder

### CODE REVIEW

File: sim/mage/talents.go [hunk 1]
```diff
@@ -403,8 +403,6 @@ func (mage *Mage) registerCombustionCD() {
 		Duration: time.Minute * 3,
 	}
 
-	fireCombCritMult := mage.SpellCritMultiplier(1, 0.1) / mage.SpellCritMultiplier(1, 0)
-
 	var fireSpells []*core.Spell
 	mage.OnSpellRegistered(func(spell *core.Spell) {
 		if spell.SpellSchool.Matches(core.SpellSchoolFire) && spell.Flags.Matches(SpellFlagMage) {
```
Comment (kayla-glick) [inline sim/mage/talents.go:408]: ❓ Mages still gain 10% crit chance per non-crit cast during combustion
image

Comment (vigo2) [reply sim/mage/talents.go:408]: Ye, but the "When activated, this spell increases your critical strike damage bonus with Fire damage spells by 50%..." part is gone, which is what this removes.

**Review by kayla-glick — COMMENTED** [review]
Racials, stats, druid, shaman, priest, warlock, hunter all look 👍 to me. I think the mage incinerate change is maybe wrong? I can't speak for warrior because I have no idea what that code does 😂

File: sim/mage/talents.go (hunk 1 above)
Comment (kayla-glick) [reply sim/mage/talents.go:408]: Oooh right CritMultiplier that's crit damage. Ok cool lgtm

**Review by TheGroxEmpire — APPROVED** [review]

**Pull request merged**